crates/spice_cloud/src/lib.rs

```diff
@@ -66,175 +79,190 @@ pub enum Error {
         source: Box<dyn std::error::Error + Sync + Send>,
     },
 
-    #[snafu(display("Spice Cloud secret not found"))]
-    SpiceSecretNotFound {},
-
     #[snafu(display("Spice Cloud api_key not provided"))]
     SpiceApiKeyNotFound {},
 
-    #[snafu(display("Unable to connect to Spice Cloud: {source}"))]
-    UnableToConnectToSpiceCloud { source: reqwest::Error },
-
     #[snafu(display("Unable to build accelerated table: {source}"))]
     UnableToBuildAcceleratedTable {
         source: AcceleratedTableBuilderError,
     },
+
+    #[snafu(display("Error exporting task_history records: {source}"))]
+    UnableToExportTaskHistoryData {
+        source: Box<dyn std::error::Error + Send + Sync>,
+    },
 }
 
-pub struct SpiceExtension {
+pub struct ScpManagementExtension {
     manifest: ExtensionManifest,
     api_key: String,
 }
 
-impl SpiceExtension {
+impl ScpManagementExtension {
     #[must_use]
     pub fn new(manifest: ExtensionManifest) -> Self {
-        SpiceExtension {
+        ScpManagementExtension {
             manifest,
             api_key: String::new(),
         }
     }
 
-    fn metrics_enabled(&self) -> bool {
-        if !self.manifest.enabled {
-            return false;
-        }
-
+    fn get_flight_url(&self) -> String {
         self.manifest
             .params
-            .get("metrics")
-            .map_or_else(|| false, |v| v.eq_ignore_ascii_case("true"))
+            .get("flight_endpoint")
+            .unwrap_or(&"https://flight.spiceai.io".to_string())
```

Comment:
Should this be like the Spice.ai data connector?
```rust
let default_flight_url: Arc<str> = if cfg!(feature = "dev") {
        "https://dev-flight.spiceai.io".into()
    } else {
        "https://flight.spiceai.io".into()
    };
```

[source](https://github.com/spiceai/spiceai/blob/726e86526d38e1c7f0b51aef66a7452be81a33c1/crates/runtime/src/dataconnector/spiceai.rs#L173-L177)